Sources/PostgresNIO/New/PostgresChannelHandler.swift

```diff
@@ -235,6 +238,56 @@ final class PostgresChannelHandler: ChannelDuplexHandler {
                 listener.failed(CancellationError())
                 return
             }
+        case .executePreparedStatement(let preparedStatement):
+            switch self.preparedStatementState.lookup(
+                name: preparedStatement.name,
+                context: preparedStatement
+            ) {
+            case .prepareStatement:
+                let promise = self.eventLoop.makePromise(of: RowDescription?.self)
+                promise.futureResult.whenSuccess { rowDescription in
+                    self.prepareStatementComplete(
+                        name: preparedStatement.name,
+                        rowDescription: rowDescription,
+                        context: context
+                    )
+                }
+                promise.futureResult.whenFailure { error in
```

Comment:
With this I want to propagate failures happening during preparation to all the pending queries we have.
I am not 100% convinced by the way it looks (especially by the need to do `error as! PSQLError`) but I cannot find any better alternative.

Do you have any advice?

Reply:
First I think we should use whenComplete here instead of whenSuccess and whenFailure. Reason for this is, we can save an allocation...

Second, we must check if the error we get is actually a PSQLError, if it is not, we need to wrap error with a PSQLError.